### PR TITLE
Prevent advert play/pause/end/progress events from being tracked by t he main video tracking when the ad library reuses the primary <video> element

### PR DESCRIPTION
**Pull request by rowanbeentje**

### CODE REVIEW

**Review by VladDubrovskis — APPROVED** [review]
LGTM just need to verify the state is set per video in playlist

File: src/js/video.js [hunk 1]
```diff
@@ -7,6 +7,16 @@ import Playlist from './playlist';
 
 function eventListener(video, ev) {
 
+	// On some platforms (eg iOS), the Google advert library will use the main <video> element
+	// used by o-video to also play a pre-roll advert; as the advert plays, this will trigger
+	// the normal <video> event listeners.  Discard events that we can identify as coming
+	// from the pre-roll rather than the main content.
+	// To do this, check whether advertising is still enabled (it'll be disabled on any error),
+	// and for the video ads load and completed flags.
+	if (video.opts.advertising && video.videoAds && video.videoAds.adsLoaded && !video.videoAds.adsCompleted) {
```
Comment (JakeChampion) [inline src/js/video.js:16]: Is there a way to include this in our test suite?

File: src/js/ads.js [hunk 2]
```diff
@@ -296,11 +305,13 @@ class VideoAds {
 	}
 
 	contentPauseRequestHandler() {
+		this.adsCompleted = false;
```
Comment (JakeChampion) [inline src/js/ads.js:308]: Could we have tests for when `this.adsCompleted` is assigned a new value?

File: src/js/video.js (hunk 1 above)
Comment (rowanbeentje) [reply src/js/video.js:16]: There don't appear to be any tests covering tracking at all, so although I looked at extending these, I'm afraid I balked at writing all the tests from scratch...

File: src/js/ads.js (hunk 2 above)
Comment (rowanbeentje) [reply src/js/ads.js:308]: This looks like it should be possible - appears Alberto set up some ads library mocking, so I can least try extending those events 👍

File: src/js/video.js (hunk 1 above)
Comment (JakeChampion) [reply src/js/video.js:16]: @lc512k would it be possible to add tests for the tracking at all?

File: src/js/ads.js (hunk 2 above)
Comment (JakeChampion) [reply src/js/ads.js:308]: @VladDubrovskis Do you think tests around the ad logic could be added?

Comment (rowanbeentje) [reply src/js/ads.js:308]: Added in https://github.com/Financial-Times/o-video/pull/65 I think

**Pull request merged**